### PR TITLE
Add extra functions from foobaz/gdal

### PR DESCRIPTION
**Pull request by peterstace**
Adds some extra files required for tiles (around GeoTIFF writing).

### CODE REVIEW

File: cpl.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	"unsafe"
+)
+
+// Get a configuration option.
+func GetConfigOption(key, def string) string {
+	k := C.CString(key)
+	d := C.CString(def)
+	defer C.free(unsafe.Pointer(k))
+	defer C.free(unsafe.Pointer(d))
+	opt := C.CPLGetConfigOption(k, d)
```
Comment (den3tsou) [inline cpl.go:24]: Just wondering what does `CPL` mean in this context?

File: gdal.go [hunk 2]
```diff
@@ -1849,3 +1847,29 @@ func VSIFReadL(nSize, nCount int, file VSILFILE) []byte {
 
 	return data
 }
+
+// Fetch buffer underlying memory file.
+func VSIGetMemFileBuffer(filename string, unlinkAndSeize bool) ([]byte, error) {
```
Comment (den3tsou) [inline gdal.go:1852]: Just wondering what is VSI?

**Review by den3tsou — COMMENTED** [review]
CGO is an alien language to me, so I wouldn't able to review it. but with few questions.

File: cpl.go (hunk 1 above)
Comment (peterstace) [reply cpl.go:24]: CPL == [Common Portability Library](https://trac.osgeo.org/gdal/wiki/FAQGeneral#WhatdoesCPLstandfor).

BTW, this function corresponds to https://www.gdal.org/cpl__conv_8h.html#a21a509890bf6632f1b1d836c1d3bb970 in C land.

File: gdal.go (hunk 2 above)
Comment (peterstace) [reply gdal.go:1852]: I _think_ VSI refers to GDAL's virtual file system (or has something to do with it). I'm having trouble finding info about it.

**Review by jh-sz — APPROVED** [review]

**Review by den3tsou — APPROVED** [review]

**Pull request merged**